### ISSUE TITLE
Incorrect linting in `exchangelib` library: example within

### ISSUE THREAD

**Issue by DeflateAwning** [microsoft/pyright#7378]
**Describe the bug**
The following code gives a linting error:

```python
from exchangelib import Credentials, Account

# from README for completeness
credentials = Credentials("john@example.com", "topsecret")
account = Account("john@example.com", credentials=credentials, autodiscover=True)

# part with linting problem:
email_message = account.inbox.filter(subject__contains='the', sender__contains='example.com').order_by('-datetime_received')[0]
```

This code runs fine, but the `.filter` part of `account.inbox.filter(...)` gets the following linting error in Pyright:

```
Cannot access member "filter" for type "threaded_cached_property"
  Member "filter" is unknown
```

**Additional context**
Python 3.9, exchangelib 5.1.0

Related to: https://github.com/ecederstrand/exchangelib/issues/1276

**VS Code extension or command-line**
VS Code

**Comment by erictraut** [comment]
Pyright isn't a linter, it's a type checker. If you're interested in linting, I recommend using pylint or ruff.

Pyright performs static type analysis and reports type violations in your code. It relies on type information supplied in libraries. The `exchangelib` library appears to be an untyped library. In other words, it lacks type annotations and doesn't contain a "py.typed" marker. Pyright will do its best to infer types in this case, but type inference can lead to incorrect types. If this library is important to you, please consider contacting the maintainers and requesting that they add type annotations to it. You also have the option of creating your own type stubs. 

For more information about type checking, please refer to [this documentation](https://microsoft.github.io/pyright/#/getting-started).

I'm going to close this issue because pyright is working as designed in this case.
